### ISSUE TITLE
DB upgrade failure from 0.8.8h to 1.1.20

### ISSUE THREAD

**Issue by devsibwarra** [Cacti/cacti#946]
I had tested an upgrade last week to 1.1.19 and it went smoothly.  Decided to go to 1.1.20 without testing and ran into this issue
```
[Fail] 
ALTER TABLE snmpagent_cache_textual_conventions
MODIFY COLUMN name VARCHAR(50) NOT NULL,
MODIFY COLUMN mib VARCHAR(50) NOT NULL,
MODIFY COLUMN type VARCHAR(50) NOT NULL,
PRIMARY KEY (`name`,`mib`,`type`)
```

The SQL statement which worked for me was
```
ALTER TABLE snmpagent_cache_textual_conventions
MODIFY COLUMN name VARCHAR(50) NOT NULL,
MODIFY COLUMN mib VARCHAR(50) NOT NULL,
MODIFY COLUMN type VARCHAR(50) NOT NULL,
ADD PRIMARY KEY (`name`,`mib`,`type`)
```

**Comment by pautiina** [comment]
Look issue #938
Already resolved on commit 7bddb90cc2c5e0ddfb8f66e914ab237b8c9db4e8

**Comment by devsibwarra** [comment]
Thanks.  My search-fu was not well trained, but I'll try to do better in the future  :)
